src/spend/secret.rs

```diff
@@ -0,0 +1,143 @@
+use super::public::PublicKey;
+use super::stealth::StealthAddress;
+
+use crate::sponge;
+use crate::{JubJubScalar, ViewKey};
+
+use dusk_jubjub::GENERATOR_EXTENDED;
+
+use std::fmt;
+
+use rand::rngs::StdRng;
+use rand::SeedableRng;
+use rand::{CryptoRng, RngCore};
+use sha2::{Digest, Sha256};
+
+/// Secret pair of `a` and `b`
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct SecretKey {
+    a: JubJubScalar,
+    b: JubJubScalar,
+}
+
+impl Default for SecretKey {
+    fn default() -> Self {
+        SecretKey {
+            a: JubJubScalar::random(&mut rand::thread_rng()),
+            b: JubJubScalar::random(&mut rand::thread_rng()),
```

Comment:
I've never seen `default` used in this way. Usually in `default` you return the default impl of your attributes. Since when you call default you usually want to generate the structure to edit it later. So it doesn't make any sense to waste time computing useless scalars.

If you want to provide a function that does this, my suggestion would be:

```rust
pub fn random<T>(rand: &mut T) -> SecretKey
    where
        T: Rng + CryptoRng,
    {
        SecretKey {
            a: JubJubScalar::random(rand),
            b: JubJubScalar::random(rand),
        }
    }
```

Reply:
I just saw that you already have it. Therefore, this should be something like:
```rust
    fn default() -> Self {
        SecretKey {
            a: JubJubScalar::default(),
            b: JubJubScalar::default(),
        }
    }
```

Reply:
This was the original implementation of `Phoenix` for the `SecretKey`. I kept because it actually made sense to me.
Notice that the fact that "Usually in `default` you return the default impl of your attributes" is a false assumption. By Rust specs, "*default values are often some kind of initial value, identity value, or anything else that may make sense as a default.*"

In case of the `SecretSpendKey` the only default values that makes sense are random scalars. If you don't want to construct a "default" `SecretSpendKey`, then you pass such values using the constructor `new`.

Reply:
Of course as soon as a scenario that invalidate this statements arise, we can definitely change it.
But I don't see atm something like that for the `SecretSpendKey` (It's not like the `PublicSpendKey`).

Reply:
It's fine then.